sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -158,28 +163,27 @@ async fn test_download_blob(ctx: TestContext) -> Result<(), Box<dyn Error>> {
 async fn test_upload_blob(ctx: TestContext) -> Result<(), Box<dyn Error>> {
     // Recording Setup
     let recording = ctx.recording();
-    let mut options = BlobClientOptions::default();
-    recording.instrument(&mut options.client_options);
-
-    // Setup
-    let endpoint = format!(
-        "https://{}.blob.core.windows.net/",
-        recording.var("AZURE_STORAGE_ACCOUNT_NAME", None).as_str()
-    );
+    let (options, endpoint) = recorded_test_setup(recording, BlobClientOptions::default()).await;
+    let container_name = recording
+        .random_string::<17>(Some("container"))
+        .to_ascii_lowercase();
+    let blob_name = recording
+        .random_string::<12>(Some("blob"))
+        .to_ascii_lowercase();
 
     // Act
     let container_client = ContainerClient::new(
         &endpoint,
-        String::from("testcontainer3"),
+        container_name.clone(),
         recording.credential(),
         Some(options.clone()),
     )?;
     container_client.create_container(None).await?;
 
     let blob_client = BlobClient::new(
         &endpoint,
-        String::from("testcontainer3"),
-        String::from("test_upload_blob.txt"),
+        container_name.clone(),
```

Comment:
nit: technically don't need this clone since it's the last use. Sort of a weird pattern to clone all but the last one but that's what you did in other tests so maybe be consistent. Or clone everywhere.